bundles/org.eclipse.swt/Eclipse SWT Drag and Drop/cocoa/org/eclipse/swt/dnd/DropTarget.java

```diff
@@ -115,30 +116,42 @@ public class DropTarget extends Widget {
 	static final String DEFAULT_DROP_TARGET_EFFECT = "DEFAULT_DROP_TARGET_EFFECT"; //$NON-NLS-1$
 	static final String IS_ACTIVE = "org.eclipse.swt.internal.isActive"; //$NON-NLS-1$
 
+	public void handleLabels(Control c, long cls) {
+		// If the content view can be an image view, then add the dragging methods to the image view too.
+	    // This is used by Label so that dragging can work even when the Label has an image set on it.
+	    if (!labelDragHandlersAdded) {
+	        return;
+	    }
+	    if (c instanceof Label) {
+	        addDragHandlers(cls);  // adding the handlers to label class
+	        long imageView = 0;
+	        if (((imageView = OS.objc_msgSend(c.view.id, OS.sel_getImageView)) != 0)) {
+	            cls = OS.object_getClass(imageView);
+	            addDragHandlers(cls);
+	            labelDragHandlersAdded = false;
```

Comment:
Should `labelDragHandlersAdded = false` be  `labelDragHandlersAdded = true` since you are setting the flag after adding the handler? The check on line 122 should be inverted as well.

Reply:
I have made the requested changes.